logfire/_internal/main.py

```diff
@@ -1990,6 +1990,7 @@ def __enter__(self) -> LogfireSpan:
                     attributes=self._otlp_attributes,
                     links=self._links,
                 )
+            self._span.__enter__()
```

Comment:
Same note as above, this might not be necessary but seems fine to add and arguably more correct(?)